services/audit-service/src/main/java/datawave/microservice/audit/replay/config/ReplayProperties.java

```diff
@@ -0,0 +1,146 @@
+package datawave.microservice.audit.replay.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+import javax.validation.constraints.PositiveOrZero;
+import java.util.concurrent.TimeUnit;
+
+@Validated
+@ConfigurationProperties(prefix = "audit.replay")
+public class ReplayProperties {
+    private boolean enabled;
+    private boolean publishEvents = true;
+    
+    @PositiveOrZero
+    private long idleTimeoutMillis = TimeUnit.SECONDS.toMillis(10);
+    
+    @PositiveOrZero
+    private long stopGracePeriodMillis = 500L;
+    
+    @PositiveOrZero
+    private long statusUpdateIntervalMillis = TimeUnit.SECONDS.toMillis(1);
+    
+    @PositiveOrZero
+    private long lockWaitTimeMillis = TimeUnit.SECONDS.toMillis(5);
+    
+    @PositiveOrZero
+    private long lockLeaseTimeMillis = TimeUnit.SECONDS.toMillis(5);
```

Comment:
Can you better describe the difference between lease/lock timeouts? I'm not clear on when I would use one over the other and the impacts

Reply:
lockWaitTimeMillis is the amount of time we will wait to acquire a lock.

lockLeaseTimeMillis is the amount of time we are allowed to hold the lock before it is automatically released.  This is meant to safeguard against cases where an audit service instance goes down while processing a locking call.  If we didn't have this lease timeout, and our task died mid-call, the lock would never be released.

Reply:
Added a description in the latest commit.